### DISCUSSION TITLE
GridMap hide floors above current floor for editing hidden parts

### DISCUSSION THREAD

**Discussion by dwidge** [godotengine/godot-proposals#8584] [Editor]
Hi, is there a way in the editor to hide all floors above the current floor so that we can see within a 3D building with many floors? We can clear away parts of the upper floors when we need to edit something on a lower floor, but have to rebuild the upper floors each time. We could use one GridMap object per floor but will that affect performance for tall buildings?

**Comment by smix8** [comment]
Using more GridMaps does not really affect performance in a worrying way.

What matters for performance is the Octant size and how many different items are placed into a single Octant.

Each Octants combines all cells with the same item into a MultiMesh, so if you have very large Octant with a single item it is peak efficient, but if you have a very small Octant with a different item placed in every single Octant cell it is peak inefficient. Also each Octant updates on changes on its own so that is also important for runtime changes.

**Reply by dwidge** [reply]
Thank you, will try use multi GridMaps.
